### ISSUE TITLE
[OpenStack] Cluster destroy fails on 1st attempt

### ISSUE THREAD

**Issue by smalleni** [openshift/installer#2424]
# Version

```console
$ openshift-install version
./openshift-install v4.2.0
built from commit 79267e5c1fb5835c694d36cad8d7b4417c1e2d65
release image quay.io/openshift-release-dev/ocp-release-nightly@sha256:49da18dcaf0dd58c2356c6a2009056c45facaca8171db6c5705ea67723523b78
```

# Platform: OpenStack

# What happened?
Installation of my ocp on osp cluster failed. I tried to clean up the install and retry. However, destroying the cluster does not work on the first attempt. It only works on the second attempt. Looks like there are some dependency/timing issues when deleting neutron networks.
```console
shiftstack) [stack@f08-h17-b01-5039ms ocp]$ ./openshift-install destroy cluster --log-level debug  --dir ostest2             
DEBUG OpenShift Installer v4.2.0                   
DEBUG Built from commit 79267e5c1fb5835c694d36cad8d7b4417c1e2d65 
DEBUG Deleting openstack networks                  
DEBUG Deleting openstack routers                   
DEBUG Deleting openstack floating ips              
DEBUG Deleting openstack containers                
DEBUG Deleting openstack trunks                    
DEBUG Deleting openstack servers                   
DEBUG Deleting openstack subnets                   
DEBUG Deleting openstack subnet-pools              
DEBUG Deleting openstack ports                     
DEBUG Deleting OpenStack volumes                   
DEBUG Deleting openstack security-groups           
DEBUG Deleting openstack load balancers            
DEBUG No suitable endpoint could be found in the service catalog. 
DEBUG Exiting deleting openstack load balancers    
DEBUG goroutine deleteLoadBalancers complete       
DEBUG Exiting deleting openstack subnet-pools
DEBUG Deleting openstack ports                     
DEBUG Deleting OpenStack volumes                   
DEBUG Deleting openstack security-groups           
DEBUG Deleting openstack load balancers            
DEBUG No suitable endpoint could be found in the service catalog. 
DEBUG Exiting deleting openstack load balancers    
DEBUG goroutine deleteLoadBalancers complete       
DEBUG Exiting deleting openstack subnet-pools      
DEBUG goroutine deleteSubnetPools complete         
DEBUG Deleting Trunk: 2529f51a-408a-472d-9e20-b80476e0aaea 
DEBUG Deleting Floating IP: d65fc8ee-fba4-4c4e-9aee-b0507568200d 
DEBUG Exiting deleting OpenStack volumes           
DEBUG goroutine deleteVolumes complete             
DEBUG Deleting Subnet: 3851d4fb-fe61-4efa-932f-dab0b73501de 
DEBUG Deleting Security Group: c2d68e51-7107-4023-a0c7-5aaad7e50463 
DEBUG Deleting object: bootstrap.ign               
DEBUG Deleting network: b624b0a3-093f-4739-9dde-9bb5a8b29256 
DEBUG Deleting container: ostest2-b4rht            
DEBUG Exiting deleting openstack containers        
DEBUG goroutine deleteContainers complete          
DEBUG Deleting Trunk failed: Expected HTTP response code [] when accessing [DELETE http://172.21.0.10:9696/v2.0/trunks/2529f51
a-408a-472d-9e20-b80476e0aaea], but got 409 instead
{"NeutronError": {"message": "Trunk 2529f51a-408a-472d-9e20-b80476e0aaea is currently in use.", "type": "TrunkInUse", "detail"
: ""}} 
DEBUG Exiting deleting openstack trunks            
DEBUG Deleting Security Group failed: Expected HTTP response code [] when accessing [DELETE http://172.21.0.10:9696/v2.0/secur
ity-groups/c2d68e51-7107-4023-a0c7-5aaad7e50463], but got 409 instead
{"NeutronError": {"message": "Security Group c2d68e51-7107-4023-a0c7-5aaad7e50463 in use.", "type": "SecurityGroupInUse", "det
ail": ""}} 
DEBUG Exiting deleting openstack security-groups   
DEBUG Deleting Floating IP: d65fc8ee-fba4-4c4e-9aee-b0507568200d 
DEBUG Deleting Subnet failed: Expected HTTP response code [] when accessing [DELETE http://172.21.0.10:9696/v2.0/subnets/3851d
4fb-fe61-4efa-932f-dab0b73501de], but got 409 instead
{"NeutronError": {"message": "Unable to complete operation on subnet 3851d4fb-fe61-4efa-932f-dab0b73501de: One or more ports h
ave an IP allocation from this subnet.", "type": "SubnetInUse", "detail": ""}} 
DEBUG Exiting deleting openstack subnets           
DEBUG Deleting Network failed: Expected HTTP response code [] when accessing [DELETE http://172.21.0.10:9696/v2.0/networks/b62
4b0a3-093f-4739-9dde-9bb5a8b29256], but got 409 instead
{"NeutronError": {"message": "Unable to complete operation on network b624b0a3-093f-4739-9dde-9bb5a8b29256. There are one or m
ore ports still in use on the network.", "type": "NetworkInUse", "detail": ""}} 
DEBUG Exiting deleting openstack networks          
FATAL Resource not found           
shiftstack) [stack@f08-h17-b01-5039ms ocp]$ ./openshift-install destroy cluster --log-level debug  --dir ostest2

This works on second attempt....

```
# What you expected to happen?
Cluster destroy should work on first attempt

# How to reproduce it (as minimally and precisely as possible)?

```console
./openshift-install destroy cluster --log-level debug  --dir ostest2

```

# Anything else we need to know?

Enter text here.

# References

- enter text here.

**Comment by smalleni** [comment]
@tomassedovic PTAL

**Comment by pierreprinetti** [comment]
#2408 should fix

**Comment by pierreprinetti** [comment]
#2408 is now merged into `master`. Do you mind trying again @smalleni?

**Comment by smalleni** [comment]
I used openshift-install-linux-4.2.0-0.nightly-2019-10-01-210901.tar.gz and seeing the same failure.

**Comment by mandre** [comment]
@smalleni at this point the patch is not included in the 4.2 branch -- we've branched off and `master` now represents the dev branch for the 4.3 release. We may consider a backport for a 4.2.x release.
Let us know if you still encounter the problem with master.

**Comment by pierreprinetti** [comment]
/close
A fix [has been backported](https://github.com/openshift/installer/pull/2701) to 4.2.
Please reopen if you still face the same issue.
